### PR TITLE
projects: add shiftfs project

### PR DESCRIPTION
**Pull request by tych0**
Here's the most recent shiftfs patches. Sounded like @estesp was interested in this as a project from our last security SIG, so here it is!

### CODE REVIEW

File: projects/shiftfs/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    # setuid 1000 unshare -rm
+    # cat /proc/self/uidmap
+             0       1000          1
+    # mkdir dest
+    # mount -t shiftfs source dest
+    # stat dest/foo | grep Uid
+    Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
+
+And thanks to the magic of shiftfs, the file is root owned in the user
+namespce.
```
Comment (riyazdf) [inline projects/shiftfs/README.md]: typo `namespace`

File: projects/shiftfs/shiftfs.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+kernel:
+  image: "linuxkitprojects/kernel-shiftfs:4.11.4-881a041fc14bd95814cf140b5e98d97dd65160b5"
```
Comment (riyazdf) [inline projects/shiftfs/shiftfs.yml]: 👍

**Comment by estesp** [comment]
cc: @jejb you might be interested

**Comment by riyazdf** [comment]
merging now but feel free to continue the conversation here or in the #linuxkit channel on the community slack

**Pull request merged**